### PR TITLE
Fix several mismatching default option values and their comments

### PR DESCRIPTION
**Pull request by aleksraiden**
In this PR:

- rocksdb.max_open_files defaults setup to 8096, but in comment old value - 4096
- Correct notice about max_write_buffer_number behavior (Misspelling, must be < 3, e.g.: If max_write_buffer_number < 3, writing will be slowed down to options.delayed_write_rate if we are writing to the last write buffer allowed.) 
- Fix max_sub_compactions  describe
- rocksdb.block_size defaults are 16384 or 16Kb, no 4 Kb (as in comment)
- rocksdb.cache_index_and_filter_blocks defaults are yes instead of no in comment

### CODE REVIEW

**Comment by PragmaTwice** [comment]
Maybe you also need to update the default value in https://github.com/apache/kvrocks/blob/unstable/src/config/config.cc .

**Comment by aleksraiden** [comment]
> Maybe you also need to update the default value in https://github.com/apache/kvrocks/blob/unstable/src/config/config.cc .

You are right, done

**Review by caipengbo — DISMISSED** [review]

**Comment by caipengbo** [comment]
@PragmaTwice Hi, Why did CI fail?

File: kvrocks.conf [hunk 1]
```diff
@@ -576,7 +576,7 @@ rocksdb.target_file_size_base 128
 # The default and the minimum number is 2, so that when 1 write buffer
 # is being flushed to storage, new writes can continue to the other
 # write buffer.
-# If max_write_buffer_number > 3, writing will be slowed down to
+# If max_write_buffer_number < 3, writing will be slowed down to
```
Comment (caipengbo) [inline kvrocks.conf]: This is greater than 3, please refer to:
https://github.com/facebook/rocksdb/blob/36f48d16a8a14fb3bb22c6d9ed5ba4ad1f01bc50/include/rocksdb/advanced_options.h#L282

Comment (aleksraiden) [reply kvrocks.conf]: Hmm, you are right, thanks! It's fully unlogical for me, but maybe this are ok )

**Review by caipengbo — DISMISSED** [review]
LGTM

File: kvrocks.conf [hunk 2]
```diff
@@ -547,7 +547,7 @@ rocksdb.row_cache_size 0
 # files opened are always kept open. You can estimate number of files based
 # on target_file_size_base and target_file_size_multiplier for level-based
 # compaction. For universal-style compaction, you can usually set it to -1.
-# Default: 4096
+# Default: 8096
```
Comment (torwig) [inline kvrocks.conf:550]: This value is 4096 by default: https://github.com/apache/kvrocks/blob/unstable/src/config/config.cc#L166

File: kvrocks.conf [hunk 3]
```diff
@@ -603,7 +603,7 @@ rocksdb.max_background_flushes -1
 # This value represents the maximum number of threads that will
 # concurrently perform a compaction job by breaking it into multiple,
 # smaller ones that are run simultaneously.
-# Default: 2 (i.e. no subcompactions)
+# Default: 2
```
Comment (torwig) [inline kvrocks.conf:606]: The default value here is 1, according to the source code: https://github.com/apache/kvrocks/blob/unstable/src/config/config.cc#L172

File: kvrocks.conf [hunk 4]
```diff
@@ -647,12 +647,12 @@ rocksdb.wal_size_limit_mb 16384
 # actual size of the unit read from disk may be smaller if
 # compression is enabled.
 #
-# Default: 4KB
+# Default: 16KB
```
Comment (torwig) [inline kvrocks.conf:650]: 4096 is OK here: https://github.com/apache/kvrocks/blob/unstable/src/config/config.cc#L165

**Comment by torwig** [comment]
Wanted to cancel my comments but failed :)

**Review by mapleFU — DISMISSED** [review]
Great work!

Seems it doesn't change the default value because the conf file will overwrite the code default?

**Review by torwig — DISMISSED** [review]

**Comment by aleksraiden** [comment]
Thanks, @torwig for helps with config.cc

File: src/config/config.cc [hunk 5]
```diff
@@ -162,22 +162,22 @@ Config::Config() {
...
+      {"rocksdb.max_sub_compactions", false, new IntField(&rocks_db.max_sub_compactions, 4, 0, 16)},
       {"rocksdb.delayed_write_rate", false, new Int64Field(&rocks_db.delayed_write_rate, 0, 0, INT64_MAX)},
       {"rocksdb.wal_ttl_seconds", true, new IntField(&rocks_db.wal_ttl_seconds, 3 * 3600, 0, INT_MAX)},
       {"rocksdb.wal_size_limit_mb", true, new IntField(&rocks_db.wal_size_limit_mb, 16384, 0, INT_MAX)},
       {"rocksdb.max_total_wal_size", false, new IntField(&rocks_db.max_total_wal_size, 64 * 4 * 2, 0, INT_MAX)},
       {"rocksdb.disable_auto_compactions", false, new YesNoField(&rocks_db.disable_auto_compactions, false)},
       {"rocksdb.enable_pipelined_write", true, new YesNoField(&rocks_db.enable_pipelined_write, false)},
       {"rocksdb.stats_dump_period_sec", false, new IntField(&rocks_db.stats_dump_period_sec, 0, 0, INT_MAX)},
-      {"rocksdb.cache_index_and_filter_blocks", true, new YesNoField(&rocks_db.cache_index_and_filter_blocks, false)},
+      {"rocksdb.cache_index_and_filter_blocks", false, new YesNoField(&rocks_db.cache_index_and_filter_blocks, true)},
```
Comment (caipengbo) [inline src/config/config.cc]: It should be readonly option

Comment (aleksraiden) [reply src/config/config.cc]: Yes, my bug, will be fixed asp, sorry

**Review by caipengbo — DISMISSED** [review]

File: src/config/config.cc [hunk 6]
```diff
@@ -162,22 +162,22 @@ Config::Config() {
...
-      {"rocksdb.max_open_files", false, new IntField(&rocks_db.max_open_files, 4096, -1, INT_MAX)},
+      {"rocksdb.block_size", true, new IntField(&rocks_db.block_size, 16384, 0, INT_MAX)},
+      {"rocksdb.max_open_files", false, new IntField(&rocks_db.max_open_files, 8096, -1, INT_MAX)},
       {"rocksdb.write_buffer_size", false, new IntField(&rocks_db.write_buffer_size, 64, 0, 4096)},
       {"rocksdb.max_write_buffer_number", false, new IntField(&rocks_db.max_write_buffer_number, 4, 0, 256)},
       {"rocksdb.target_file_size_base", false, new IntField(&rocks_db.target_file_size_base, 128, 1, 1024)},
       {"rocksdb.max_background_compactions", false, new IntField(&rocks_db.max_background_compactions, 2, -1, 32)},
       {"rocksdb.max_background_flushes", true, new IntField(&rocks_db.max_background_flushes, 2, -1, 32)},
-      {"rocksdb.max_sub_compactions", false, new IntField(&rocks_db.max_sub_compactions, 1, 0, 16)},
+      {"rocksdb.max_sub_compactions", false, new IntField(&rocks_db.max_sub_compactions, 4, 0, 16)},
```
Comment (PragmaTwice) [inline src/config/config.cc]: In kvrocks.conf it is 2 instead of 4.

Comment (aleksraiden) [reply src/config/config.cc]: Fixed, thanks

**Review by git-hulk — APPROVED** [review]

**Review by torwig — APPROVED** [review]

**Review by PragmaTwice — APPROVED** [review]

**Pull request merged**